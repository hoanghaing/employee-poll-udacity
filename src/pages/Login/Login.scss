.container {
  background-color: rgb(227, 227, 227);
  height: 100%;

  justify-content: center;
  align-items: center;
}