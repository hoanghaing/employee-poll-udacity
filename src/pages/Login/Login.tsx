import React from 'react';
import './Login.scss';
import * as api from '@/api/index';
const Login = () => {
  const subMitUser = {};
  //  = api._getUsers().then((res) => {
  //   console.log(res);
  // });
  return (<div className="container flex flex-col gap-[12px]">
    <div className="hero min-h-screen bg-base-200">
      <div className="hero-content flex-col lg:flex-row-reverse">
        <div className="text-center lg:text-left">
          <h1 className="text-5xl font-bold">Login now!</h1>
          <p className="py-6">Trusted by Udacity, best software for creating poll. Free for all user until 10 poll</p>
        </div>
        <div className="card shrink-0 w-full max-w-sm shadow-2xl bg-base-100">
          <form className="card-body">
            <div className="form-control">
              <label className="label">
                <span className="label-text">Username</span>
              </label>
              <input type="email" placeholder="username" className="input input-bordered" required />
            </div>
            <div className="form-control">
              <label className="label">
                <span className="label-text">Password</span>
              </label>
              <input type="password" placeholder="password" className="input input-bordered" required />
            </div>
            <div className="form-control mt-6">
              <button className="btn btn-primary">Login</button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>)
}
export default Login