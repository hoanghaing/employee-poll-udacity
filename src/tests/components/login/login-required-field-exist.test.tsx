import { expect, describe, it, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import Login from '@/pages/Login/Login';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  // Mock your Redux state here if needed
});

test('renders login form correctly', async () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
