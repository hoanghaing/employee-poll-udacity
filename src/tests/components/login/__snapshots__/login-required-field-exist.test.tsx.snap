// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders login form correctly 1`] = `
<div
  class="container flex flex-col gap-[12px]"
>
  <div
    class="hero min-h-screen bg-base-200"
  >
    <div
      class="hero-content flex-col lg:flex-row-reverse"
    >
      <div
        class="text-center lg:text-left"
      >
        <h1
          class="text-5xl font-bold"
        >
          Login now!
        </h1>
        <p
          class="py-6"
        >
          Trusted by Udacity, best software for creating poll. Free for all user until 10 poll
        </p>
      </div>
      <div
        class="card shrink-0 w-full max-w-sm shadow-2xl bg-base-100"
      >
        <form
          class="card-body"
        >
          <div
            class="form-control"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
              >
                Username
              </span>
            </label>
            <input
              class="input input-bordered"
              id="usernameInput"
              placeholder="username"
              required=""
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
              >
                Password
              </span>
            </label>
            <input
              class="input input-bordered"
              id="passwordInput"
              placeholder="password"
              required=""
              type="password"
            />
          </div>
          <div
            class="form-control mt-6"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
